Bottom: 182bd1fa512fae210f379b908daa72d366639f15
Top:    12aa9f439735ec9e0e8e3f7f6340b0af278c4fb1
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-22 16:44:07 -0400

Refresh of stac9766-codec

---

diff --git a/sound/soc/codecs/stac9766.c b/sound/soc/codecs/stac9766.c
index 9f9dfa8..31ffee3 100644
--- a/sound/soc/codecs/stac9766.c
+++ b/sound/soc/codecs/stac9766.c
@@ -454,37 +454,17 @@ struct snd_soc_codec_device soc_codec_dev_stac9766 =
 };
 EXPORT_SYMBOL_GPL(soc_codec_dev_stac9766);
 
-static int __init stac9766_probe(struct platform_device *pdev)
+static int __init stac9766_modinit(void)
 {
-	snd_soc_register_dais(stac9766_dai, ARRAY_SIZE(stac9766_dai));
-#if defined(CONFIG_SND_SOC_OF_SIMPLE)
-	/* Tell the of_soc helper about this codec */
-	of_snd_soc_register_codec(&soc_codec_dev_stac9766, pdev->dev.archdata.of_node,
-									stac9766_dai, ARRAY_SIZE(stac9766_dai),
-									pdev->dev.archdata.of_node);
-#endif
-	return 0;
+	return snd_soc_register_dais(stac9766_dai, ARRAY_SIZE(stac9766_dai));
 }
+module_init(stac9766_modinit);
 
-static struct platform_driver stac9766_driver =
-{
-	.probe = stac9766_probe,
-	.driver = {
-			.name = "stac9766",
-	},
-};
-
-static __init int stac9766_driver_init(void)
+static void __exit stac9766_exit(void)
 {
-	return platform_driver_register(&stac9766_driver);
+	snd_soc_unregister_dais(stac9766_dai, ARRAY_SIZE(stac9766_dai));
 }
-
-static __exit void stac9766_driver_exit(void)
-{
-}
-
-module_init(stac9766_driver_init);
-module_exit(stac9766_driver_exit);
+module_exit(stac9766_exit);
 
 MODULE_DESCRIPTION("ASoC stac9766 driver");
 MODULE_AUTHOR("Jon Smirl <jonsmirl@gmail.com>");
